### PR TITLE
Device Datasource Implementation

### PR DESCRIPTION
**Pull request by Krunal-Thakkar**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Backward compatibility is not broken

# How Has This Been Tested?
image

### CODE REVIEW

**Review by VamsiSiddu-7 — APPROVED** [review]

**Review by khareRajshree — APPROVED** [review]

File: types/v1/types.go [hunk 1]
```diff
@@ -503,18 +503,97 @@ type SdsPort struct {
 	SdsPort string `json:"sdsPort"`
 }
 
-// Device defines struct for Device
+// Device defines struct of Device for Pflex Array
```
Comment (shenda1) [inline types/v1/types.go:506]: PowerFlex

File: device.go [hunk 2]
```diff
@@ -104,5 +104,62 @@ func (sp *StoragePool) FindDevice(
...
+		device.Device.ID)
+
+	var deviceResult types.Device
+	err := device.client.getJSONWithRetry(
+		http.MethodGet, path, nil, &deviceResult)
+	if err != nil {
+		return nil, err
+	}
+
+	return &deviceResult, nil
```
Comment (rounak-adhikary) [inline device.go:164]: Getting a device from an existing device seems a bit odd. Can we make the function signature a bit like the following?
```suggestion
func (system *System) GetDevice(id string) (*types.Device, error) {

	defer TimeSpent("GetDevice", time.Now())

	path := fmt.Sprintf(
		"/api/instances/Device::%v",
		id)

	var deviceResult types.Device
	err := system.client.getJSONWithRetry(
		http.MethodGet, path, nil, &deviceResult)
	if err != nil {
		return nil, err
	}

	return &deviceResult, nil
```

Comment (Krunal-Thakkar) [reply device.go]: Updated.

File: types/v1/types.go (hunk 1 above)
Comment (Krunal-Thakkar) [reply types/v1/types.go]: Updated.

**Review by rounak-adhikary — APPROVED** [review]

**Review by AnikaAgiwal2711 — APPROVED** [review]

**Pull request merged**